app/src/main/java/edu/usf/sas/pal/muser/util/EventUtils.kt

```diff
@@ -0,0 +1,115 @@
+        val currentTimeMS = System.currentTimeMillis()
+        val nanoTime = System.nanoTime()
+        val songData = SongData(song, context)
+        val seekPositionMs = position
+        return UiEvent(uiEventType = capturedUiAction, currentTimeMs = currentTimeMS,
+                nanoTime = nanoTime, seekPositionMs = seekPositionMs, song = songData)
+    }
+
+    /**
+     * Function to populate the UiAction class with Album data when the overflow button is clicked
+     * @param album - The Album for which the action was performed
+     * @param capturedUiAction = The action that was captured.
+     * @return UiAction object
+     */
+    @JvmStatic
+    fun newUiAlbumEvent(album: Album, capturedUiAction: UiEventType): UiEvent {
+        val currentTimeMS = System.currentTimeMillis()
+        val nanoTime = System.nanoTime()
+        val albumData = AlbumData(album)
+        return UiEvent(uiEventType = capturedUiAction, currentTimeMs = currentTimeMS,
+                nanoTime = nanoTime, album = albumData)
+    }
+
+    /**
+     * Function to populate the UiAction class with AlbumArtist data when the overflow button is clicked
+     * @param albumArtist - The Album Artist for which the action was performed
+     * @param capturedUiAction - The action that was captured.
+     * @return UiAction object
+     */
+
+    @JvmStatic
+    fun newUiAlbumArtistEvent(albumArtist: AlbumArtist, capturedUiAction: UiEventType): UiEvent {
+        val currentTimeMS = System.currentTimeMillis()
+        val nanoTime = System.nanoTime()
+        val albums = mutableListOf<AlbumData>()
+        albumArtist.albums.forEach{
+            val albumData = AlbumData(it.id, it.name, it.artists, it.albumArtistName, it.year,
+                             it.numSongs, it.numDiscs, it.dateAdded, it.paths)
+            albums.add(albumData)
+        }
+        val albumArtistData = AlbumArtistData(albumArtist, albums)
+        return UiEvent(uiEventType = capturedUiAction, currentTimeMs = currentTimeMS,
+                nanoTime = nanoTime, albumArtist = albumArtistData)
+    }
+
+    /**
+     * Function to populate the UiAction class with Genre data when the overflow button is clicked
+     * @param genre - The Genre for which the action was performed.
+     * @param capturedUiAction - The action that was captured
+     * @return UiAction object
+     */
+    @JvmStatic
+    fun newUiGenreEvent(genre: Genre, capturedUiAction: UiEventType): UiEvent {
+        val currentTimeMS = System.currentTimeMillis()
+        val nanoTime = System.nanoTime()
+        val genreData = GenreData(genre)
+        return UiEvent(uiEventType = capturedUiAction, currentTimeMs = currentTimeMS,
+                nanoTime = nanoTime, genre = genreData)
+    }
+
+    /**
+     * function to get the seek position of the track.
+     * @return current seek position. 0 if there's an error
+     */
+    val position: Long
```

Comment:
Could you move this method to MusicServiceConnectionUtils as well? It really should also be in the form of a `fun xxx(` and not a variable with a `get()` function within.